client/src/app/party-select/party-select.component.ts

```diff
@@ -44,13 +46,45 @@ export class PartySelectComponent implements OnInit {
   changeParty(index: number): void {
     this.hideUnimplementedJobs = false;
     this.jobPickerVisible = true;
-        this.jobPickerModalRef.afterClose.pipe(take(1)).subscribe(
-          (jobChosen: string) => {
-            if (jobChosen) {
-              this.dpsService.party[index] = jobChosen;
-              this.dpsService.updateAllStats();
-            }
-          });
+    this.selectableRoles = this.getAvailableRoles(
+        jobConsts.JOBS.get(this.dpsService.selfJob) || jobConsts.SCHOLAR,
+        index);
+    this.jobPickerModalRef.afterClose.pipe(take(1)).subscribe(
+      (jobChosen: string) => {
+        if (jobChosen) {
+          this.dpsService.party[index] = jobChosen;
+          this.dpsService.updateAllStats();
+        }
+      });
+  }
+
+  private getAvailableRoles(selfJob: jobConsts.Job, index: number):
```

Comment:
i think this logic can actually get simplified a bit. i can see a couple options:

option 1:
* if selfjob is a tank, index into the array `[T, H, H, D, D, D, D]`
* if healer, index into `[T, T, H, D, D, D, D]`
* if dps, index into `[T, T, H, H, D, D, D]`

option 2:
* if the index is 0: TANK
* if it's 1: if selfjob is a tank, HEALER. else TANK
* if it's 2: HEALER
* if it's 3: if selfjob is a dps, HEALER. else TANK
* if it's 4-6: DPS

option 3:
consider the full standard party comp: `[T, T, H, H, D, D, D, D]`. delete the element at index 0, 2, or 4 (based on your role) and index into the resulting 7-element array.

for options 1 and 3, i think you could get rid of `getAvailableRoles` entirely, have a `partyRoles` variable and in `changeSelf`, if you're changing your job you set the `partyRoles` variable accordingly since the choice of your job determines the remaining roles in a standard comp. then you can just index into `partyRoles` instead of calling `getAvailableRoles`

thoughts? i personally think option 1 is easiest to read and understand, though it does have the downside of hard-coding the three arrays.

Reply:
Going with option 1.  I don't like the three arrays either, but you're right, it is the most readable.  Can't get rid of getAvailableRoles though.  The player and party share a job picker and it's easier to change the value of what the picker loops through than use two different variables for it.